sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,575 @@
+        optional. If not provided, a warning is raised about the future change in
+        the default which will occur after September 2021.
+
+    See Also
+    --------
+    StatsForecastARCH
+    StatsForecastGARCH
+
+    References
+    ----------
+    .. [1] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [2] Documentation of arch package (soft dependency). Forecasting Volatility with
+       ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline  # doctest: +SKIP
+    >>> from sktime.forecasting.arch import ARCH  # doctest: +SKIP
+    >>> y = load_airline()  # doctest: +SKIP
+    >>> forecaster = ARCH()  # doctest: +SKIP
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    ARCH(...)
+    >>> y_pred = forecaster.predict(fh=1)  # doctest: +SKIP
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+        "python_dependencies": "arch",
+        "capability:pred_int": True,
+    }
+
+    def __init__(
+        self,
+        mean="Constant",
+        lags=0,
+        vol="GARCH",
+        p=1,
+        o=0,
+        q=1,
+        power=2.0,
+        dist="Normal",
+        hold_back=None,
+        rescale=False,
+        update_freq=0,
+        disp=False,
+        starting_values=None,
+        cov_type="robust",
+        show_warning=False,
+        first_obs=None,
+        last_obs=None,
+        tol=None,
+        options=None,
+        backcast=None,
+        params=None,
+        start=None,
+        align="origin",
+        method="simulation",
+        simulations=10,
+        rng=None,
+        random_state=None,
+        reindex=False,
+    ):
+        self.mean = mean
+        self.lags = lags
+        self.vol = vol
+        self.p = p
+        self.o = o
+        self.q = q
+        self.power = power
+        self.dist = dist
+        self.hold_back = hold_back
+        self.rescale = rescale
+        self.update_freq = update_freq
+        self.disp = disp
+        self.starting_values = starting_values
+        self.cov_type = cov_type
+        self.show_warning = show_warning
+        self.first_obs = first_obs
+        self.last_obs = last_obs
+        self.tol = tol
+        self.options = options
+        self.backcast = backcast
+        self.params = params
+        self.start = start
+        self.align = align
+        self.method = method
+        self.simulations = simulations
+        self.rng = rng
+        self.random_state = random_state
+        self.reindex = reindex
+
+        if self.mean in ["ARX", "HARX"]:
+            ARCH._tags["ignores-exogeneous-X"] = False
+        else:
+            ARCH._tags["ignores-exogeneous-X"] = True
```

Comment:
I'll suggest if you do not modify this internal properties yourself, and instead use something along these lines:

https://github.com/sktime/sktime/blob/807a9efeb998406e9152067742f7ed357cbe7f65/sktime/forecasting/base/adapters/_generalised_statsforecast.py#L41

Reply:
ah yes, I missed that. This would change the tag for all instances and possibly create problems in larger deployments by side effects.

Reply:
Modified it.